cray/modules/ims/openapi.yaml

```diff
@@ -2232,10 +2232,8 @@ components:
           minLength: 1
         kernel_file_name:
           description: Name of the kernel file to extract and upload to the artifact repository from the /boot directory of the image root.
-          default: vmlinuz
           example: vmlinuz
           type: string
-          minLength: 1
```

Comment:
I'm curious -- why are we removing the minimum length requirement? Are we wanting them to be able to specify the empty string (rather than just omitting the argument)?

Reply:
We want to be able to allow an optional here so that they can pass in if they need to. Otherwise defaults are handled in IMS

Reply:
The minimum length only applies if they have passed the option. If they omit `kernel_file_name` entirely, the length restriction isn't applied. As long as this parameter is not marked as required, it's fine for it to still have the minLength of 1. Otherwise you are saying it is legal for them to pass in `kernel_file_name` with a value of `""`

Reply:
In other words, I assume we want IMS to give an error if someone passes in an empty string for the kernel file name, so we should leave the minimum length in place. It's fine to not pass in the file name and let it take a default value, but that's different than passing in a file name with a blank value. Does that make sense?

Reply:
That makes sense, I originally thought that the minLengh would be performing validation as well and wasn't an optional. In that case I can revert that minimum length back.

Reply:
Minimum length IS validated. If you pass in a string for the kernel file name, it has to be at least 1 character long.

If you don't pass in a string, then there is nothing for it to validate.

This is why you were able to not specify a kernel file name even before this change -- not specifying a file name is not the same as specifying a 0 length filename.

IMO we should restore the minimum length to the API spec, here and in the IMS repo, since previously passing in a 0-length string would have resulted in an error. No reason to change that here, IMO.

Reply:
I agree with this. You should only allow an empty string here if you intend to allow the user to explicitly set the empty string, which should be different from the user taking the default by not specifying the option at all. That is a pretty fundamental command line interface design principle: what the user specifies by providing an option or argument goes and overrides any default value. If it is invalid, the error checking (in this case the minimum length) should catch it.

Reply:
Will update both IMS and this PR to reflect these changes, thank you both for feedback!